lib/common/common/src/save_on_disk.rs

```diff
@@ -85,21 +85,27 @@ impl<T: Serialize + for<'de> Deserialize<'de> + Clone> SaveOnDisk<T> {
     where
         F: Fn(&T) -> bool,
     {
-        let start = std::time::Instant::now();
-        while start.elapsed() < timeout {
+        let deadline = std::time::Instant::now()
+            .checked_add(timeout)
+            .unwrap_or_else(std::time::Instant::now);
```

Comment:
How about a [saturating add](https://doc.rust-lang.org/std/time/struct.Duration.html#method.saturating_add) here?

Reply:
You're right.